feature/gribi/mpls_in_udp/otg_tests/mpls_in_udp_scale/mpls_in_udp_scale_test.go

```diff
@@ -0,0 +1,1458 @@
+
+		case deviations.GRIBIMACOverrideStaticARPStaticRoute(dut):
+			nh, op := gribi.NHEntry(nhIDs[i], "MACwithInterface", deviations.DefaultNetworkInstance(dut), fluent.InstalledInFIB, &gribi.NHOptions{Interface: port.Name(), Mac: dstMac, Dest: dstIP})
+			nhEntries = append(nhEntries, nh)
+			nhOps = append(nhOps, op)
+
+		default:
+			nh, op := gribi.NHEntry(nhIDs[i], "MACwithInterface", deviations.DefaultNetworkInstance(dut), fluent.InstalledInFIB, &gribi.NHOptions{Interface: port.Name(), Mac: dstMac})
+			nhEntries = append(nhEntries, nh)
+			nhOps = append(nhOps, op)
+		}
+	}
+
+	// Build NHG with both next-hops (ECMP)
+	nhMap := map[uint64]uint64{nhIDs[0]: 1, nhIDs[1]: 1}
+	nhg, nhgOp := gribi.NHGEntry(nhgID, nhMap, deviations.DefaultNetworkInstance(dut), fluent.InstalledInFIB)
+	nhEntries = append(nhEntries, nhg)
+	nhOps = append(nhOps, nhgOp)
+
+	// Install all NH + NHG entries
+	c.AddEntries(t, nhEntries, nhOps)
+
+	// Add IPv6 route for outer destination to point to the NHG
+	c.AddIPv6(t, outerIPv6Dst+routeV6PrfLen, nhgID, deviations.DefaultNetworkInstance(dut), deviations.DefaultNetworkInstance(dut), fluent.InstalledInFIB)
+
+	t.Logf("Installed ECMP route %s via ports 3 and 4", outerIPv6Dst+routeV6PrfLen)
+}
+
+// programMPLSinUDPEntries programs gRIBI entries for MPLS-in-UDP encapsulation. It installs a single NextHop that performs MPLS-in-UDP encapsulation with the provided outer IPv6 and UDP header attributes, associates multiple NextHopGroups (NHGs) with that NextHop, and finally installs IPv6 /128 routes pointing to each NHG.
+func programMPLSinUDPEntries(t *testing.T, dut *ondatra.DUTDevice, nextHopID, mplsLabelStart uint64, numNHGs int, outerIPv6Src, outerIPv6Dst string, outerDstUDPPort uint16, outerTTL, outerDSCP uint8) []fluent.GRIBIEntry {
+	t.Helper()
+	// Preallocate exact capacity to avoid reallocation
+	totalEntries := 1 + 2*numNHGs
+	entries := make([]fluent.GRIBIEntry, 0, totalEntries)
+
+	// Create the single MPLS-in-UDP NextHop
+	entries = append(entries,
+		fluent.NextHopEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).WithIndex(nextHopID).
+			AddEncapHeader(fluent.MPLSEncapHeader().WithLabels(mplsLabelStart),
+				fluent.UDPV6EncapHeader().
+					WithSrcIP(outerIPv6Src).
+					WithDstIP(outerIPv6Dst).
+					WithDstUDPPort(uint64(outerDstUDPPort)).
+					WithIPTTL(uint64(outerTTL)).
+					WithDSCP(uint64(outerDSCP)),
+			),
+	)
+
+	// Pre-parse base IP once
+	ipInt := big.NewInt(0).SetBytes(net.ParseIP(baseIPv6).To16())
+	tmpBytes := make([]byte, 16)
+
+	// Create NextHopGroups and IPv6 routes
+	for i := range numNHGs {
+		nhgID := nextHopGroupID + uint64(i)
+
+		// Add NHG
+		entries = append(entries, fluent.NextHopGroupEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).WithID(nhgID).AddNextHop(nextHopID, 1))
+
+		// Compute next prefix
+		ipInt.Add(ipInt, big.NewInt(1))
+		ipBytes := ipInt.FillBytes(tmpBytes)
+		prefix := fmt.Sprintf("%s%s", net.IP(ipBytes), routeV6PrfLen)
```

Comment:
This logic overlaps the helper you created in the iputil library.  Delete here and just use helper.


```suggestion
	// Pre-parse base IP once
	baseIPv6 := net.ParseIP(baseIPv6).To16()
	
	// Create NextHopGroups and IPv6 routes
	for i := range numNHGs {
		nhgID := nextHopGroupID + uint64(i)

		// Add NHG
		entries = append(entries, fluent.NextHopGroupEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).WithID(nhgID).AddNextHop(nextHopID, 1))

		// Compute next prefix
		offset := iputil.GenerateIPv6WithOffset(baseIPv6, i)
		prefix := fmt.Sprintf("%s/%s", offset.String(), routeV6PrfLen)
```